app/services/definitions/definitions.py

```diff
@@ -0,0 +1,325 @@
+import asyncio
+from typing import List, Optional
+
+from aries_cloudcontroller import (
+    AcaPyClient,
+    CredentialDefinitionSendRequest,
+    SchemaGetResult,
+    SchemaSendRequest,
+)
+
+from app.exceptions import (
+    CloudApiException,
+    handle_acapy_call,
+    handle_model_with_validation,
+)
+from app.models.definitions import (
+    CreateCredentialDefinition,
+    CreateSchema,
+    CredentialDefinition,
+    CredentialSchema,
+)
+from app.routes.trust_registry import (
+    get_schema_by_id as get_trust_registry_schema_by_id,
+)
+from app.routes.trust_registry import get_schemas as get_trust_registry_schemas
+from app.services.definitions.credential_definition_publisher import (
+    CredentialDefinitionPublisher,
+)
+from app.services.definitions.schema_publisher import SchemaPublisher
+from app.services.trust_registry.schemas import register_schema
+from app.services.trust_registry.util.issuer import assert_valid_issuer
+from app.util.assert_public_did import assert_public_did
+from app.util.definitions import (
+    credential_definition_from_acapy,
+    credential_schema_from_acapy,
+)
+from app.util.transaction_acked import wait_for_transaction_ack
+from shared.constants import GOVERNANCE_AGENT_URL
+from shared.log_config import get_logger
+
+logger = get_logger(__name__)
+
+
+async def create_schema_service(
+    aries_controller: AcaPyClient,
+    schema_request: SchemaSendRequest,
+    schema: CreateSchema,
+) -> CredentialSchema:
+    """
+    Create a schema and register it in the trust registry
+    """
+    bound_logger = logger.bind(body=schema)
+
+    publisher = SchemaPublisher(controller=aries_controller, logger=logger)
+
+    try:
+        result = await publisher.publish_schema(schema_request)
+    except CloudApiException as e:
+        if "already exist" in e.detail and e.status_code == 400:
+            result = await publisher.handle_existing_schema(schema)
+        else:
+            bound_logger.warning(
+                f"An unhandled Exception was caught while publishing schema: {e.detail}"
+            )
+            raise CloudApiException("Error while creating schema.") from e
+
+    if result.sent and result.sent.schema_id:
+        await register_schema(schema_id=result.sent.schema_id)
+    else:
+        bound_logger.error("No SchemaSendResult in `publish_schema` response.")
+        raise CloudApiException(
+            "An unexpected error occurred: could not publish schema."
+        )
+
+    result = credential_schema_from_acapy(result.sent.var_schema)
+    bound_logger.info("Successfully published and registered schema.")
+    return result
```

Comment:
I think this method needs some refactoring.
- First, the method name can just be `create_schema`, instead of `create_schema_service`.
- We want to make sure that only governance can create_schemas. Even if we know that check is happening in the route logic, there should be an assertion here as well, to check that aries_controller is indeed the governance controller, and it can't accidentally be called elsewhere as a tenant.
- Also, I think the whole try publish, if already exists call handle existing, and if result, publish to trust registry, I think that can all be part of the SchemaPublisher. So the SchemaPublisher has a .create_schema method, which does all of this, and this definitions method just forwards the request to the schema publisher, and does the `credential_schema_from_acapy` conversion. And maybe the SchemaSendRequest model creation too. Something like that. That way, `handle_existing_schema` can be an internal method (`_handle_existing_schema`), not called from outside.

Lastly, I'm wondering why the method takes both a `SchemaSendRequest` and a `CreateSchema` in its argument ... `CreateSchema` is the model we receive as request body, and SchemaSendRequest is what we send to ACA-Py. It feels like only one of those is needed here. Lots of ways to skin a cat. Maybe just move the SchemaSendRequest model creation here. Or modify handle_existing_schema to not need the CreateSchema body ... see what works best!